diff --git a/src/main.rs b/src/main.rs
index 3692849..87d7e76 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -6,6 +6,7 @@ use std::time::Duration;
 use std::{fmt, io, mem, process, slice, thread};
 
 use argh::FromArgs;
+use nix::sys::ioctl::ioctl_param_type;
 use nix::{ioctl_write_int, ioctl_write_ptr};
 
 /// Force-feedback device path.
@@ -63,7 +64,7 @@ impl Vibrator {
     }
 
     /// Stop vibration and remove effect from device.
-    fn stop(&mut self, effect_id: u64) -> Result<(), String> {
+    fn stop(&mut self, effect_id: ioctl_param_type) -> Result<(), String> {
         let fd = self.device.as_raw_fd();
         match unsafe { remove_effect(fd, effect_id) } {
             Ok(_) => Ok(()),
@@ -116,7 +117,7 @@ impl Vibrator {
         // Remove effect after it finished playing.
         let duration = Duration::from_millis(((length + interval) * count) as u64);
         thread::sleep(duration);
-        self.stop(effect.id as u64)?;
+        self.stop(effect.id as ioctl_param_type)?;
 
         Ok(())
     }

