--- a/meson.build
+++ b/meson.build
@@ -1,5 +1,5 @@
 project('gnome-shell', 'c',
-  version: '49.mobile.0',
+  version: '49.0',
   meson_version: '>= 1.3.0',
   license: 'GPL-2.0-or-later',
 )
@@ -24,7 +24,7 @@ gcr_req = '>= 3.90.0'
 gio_req = '>= 2.86.0'
 gjs_req = '>= 1.85.90'
 gtk_req = '>= 4.0'
-mutter_req = '= 49.mobile.0'
+mutter_req = '= 49.0'
 polkit_req = '>= 0.100'
 schemas_req = '>= 49.alpha'
 systemd_req = '>= 246'
