diff --git a/initrc b/initrc
index 47c6ba7..d1617a1 100644
--- a/initrc
+++ b/initrc
@@ -28,7 +28,7 @@ catacomb msg bind-key --trigger press '*' XF86PowerOff bash -c \
 
 ## Turn off display when pressing the power button
 catacomb msg bind-key --trigger release '*' XF86PowerOff bash -c \
-    "if pkill -xf -SIGINT \"sleep 0 0 0 0 0 0 0 0.5\"; then \
+    "if pkill -SIGINT -xf \"sleep 0 0 0 0 0 0 0 0.5\"; then \
         if [ \"\$(catacomb msg dpms)\" == \"on\" ]; then \
             catacomb msg dpms off; \
         else \
@@ -38,7 +38,7 @@ catacomb msg bind-key --trigger release '*' XF86PowerOff bash -c \
 
 ## Take a screenshot on `power + volume down` and save it to /tmp/screenshot.png
 catacomb msg bind-key --trigger press '*' XF86PowerOff+XF86AudioLowerVolume bash -c \
-    "pkill -xf -SIGINT \"sleep 0 0 0 0 0 0 0 0.5\"; \
+    "pkill -SIGINT -xf \"sleep 0 0 0 0 0 0 0 0.5\"; \
     tremor 150 0 1; \
     geometry=\$(slurp 2>&1); \
     if [[ \$geometry == \"selection cancelled\" ]]; then \
@@ -73,5 +73,8 @@ swayidle -w \
     timeout 210 'systemctl suspend' \
         after-resume 'catacomb msg dpms on' &
 
+# Set default postmarketOS wallpaper
+tabula --image /usr/share/wallpapers/postmarketos.jpg &
+
 # Wait for completion
 wait

